frontend/src/metabase/search/containers/SearchApp.unit.spec.tsx

```diff
@@ -0,0 +1,212 @@
+      expect(
+        screen.getByText("There weren't any results for your search."),
+      ).toBeInTheDocument();
+      expect(screen.queryByLabelText("pagination")).not.toBeInTheDocument();
+    });
+
+    it("renders search results when there is at least one result", async () => {
+      await setup({ searchText: "Card" });
+
+      const searchResultsHeader = screen.getByText('Results for "Card"');
+      expect(searchResultsHeader).toBeInTheDocument();
+
+      const searchResults = screen.getAllByTestId("search-result-item");
+      expect(searchResults.length).toEqual(1);
+      expect(screen.queryByLabelText("pagination")).not.toBeInTheDocument();
+    });
+
+    it("renders search results and pagination when there is more than PAGE_SIZE results", async () => {
+      await setup({ searchText: "a" });
+      const getPaginationTotal = () => screen.getByTestId("pagination-total");
+      const getPagination = () => screen.getByLabelText("pagination");
+      const getNextPageButton = () => screen.getByTestId("next-page-btn");
+      const getPreviousPageButton = () =>
+        screen.getByTestId("previous-page-btn");
+
+      expect(getPaginationTotal()).toHaveTextContent("5");
+      expect(getPreviousPageButton()).toBeDisabled();
+      expect(getNextPageButton()).toBeEnabled();
+      expect(getPagination()).toHaveTextContent("1 - 3");
+
+      // test next page button
+      userEvent.click(getNextPageButton());
+      await waitFor(() => {
+        expect(screen.queryByTestId("loading-spinner")).not.toBeInTheDocument();
+      });
+      expect(getPaginationTotal()).toHaveTextContent("5");
+      expect(getPreviousPageButton()).toBeEnabled();
+      expect(getNextPageButton()).toBeDisabled();
+
+      expect(getPagination()).toHaveTextContent("4 - 5");
+
+      // test previous page button
+      userEvent.click(getPreviousPageButton());
+      await waitFor(() => {
+        expect(screen.queryByTestId("loading-spinner")).not.toBeInTheDocument();
+      });
+      expect(getPaginationTotal()).toHaveTextContent("5");
+      expect(getPreviousPageButton()).toBeDisabled();
+      expect(getNextPageButton()).toBeEnabled();
+      expect(getPagination()).toHaveTextContent("1 - 3");
+    });
+  });
+
+  describe("filtering search results with the sidebar", () => {
+    it("should reload with filtered searches when a type on the right sidebar is clicked without changing URL", async () => {
+      const { history } = await setup({
+        searchText: "Card",
+      });
+
+      let url = history?.getCurrentLocation();
+      const { pathname: prevPathname, search: prevSearch } = url ?? {};
+
+      expect(screen.getAllByTestId("type-sidebar-item")).toHaveLength(2);
+
+      const expectedSidebarLabels = ["All items", "Questions"];
+      screen.getAllByTestId("type-sidebar-item").forEach(item => {
+        expect(expectedSidebarLabels).toContain(item.textContent);
+      });
```

Comment:
Nit: since there are only 2 items here, I think you could try to match it manually without looping or at least use the index from the loop to get the expected result would be better, since this doesn't guarantee that `All items` would come first.

Reply:
I've changed the test to check whether the first item in the sidebar list is `All items` in the two tests that check for this